### PR TITLE
Remove temporary files after tests execution

### PR DESCRIPTION
**Pull request by php-coder**
Some tests left temporary files after its execution. I attempted to fix some of them.

Example:
```console
$ ls -ld /tmp/gitauth*
ls: cannot access /tmp/gitauth*: No such file or directory
$ ./hack/test-integration.sh TestNewAppSourceAuthRequired
ok      TestNewAppSourceAuthRequired
./hack/test-integration.sh took 19 seconds
Complete
$ ls -ld /tmp/gitauth*
drwx------. 5 coder coder 100 Nov  9 16:42 /tmp/gitauth003198106
drwx------. 5 coder coder 100 Nov  9 16:42 /tmp/gitauth776832470
drwx------. 5 coder coder 100 Nov  9 16:42 /tmp/gitauth968712578
drwx------. 5 coder coder 100 Nov  9 16:42 /tmp/gitauth993381998
```

PTAL @mfojtik

### CODE REVIEW

File: test/integration/newapp_test.go [hunk 1]
```diff
@@ -1711,7 +1712,7 @@ func TestNewAppSourceAuthRequired(t *testing.T) {
 	}
 }
 
-func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) string {
+func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) (string, func()) {
 	// Create test directories
 	testDir, err := ioutil.TempDir("", "gitauth")
```
Comment (php-coder) [inline test/integration/newapp_test.go]: Good idea! But as I can see other integration tests don't use it.

Could you prepare PR for that or fill the issue, because I'm fixing a different problem here.

Comment (stevekuznetsov) [inline test/integration/newapp_test.go]: `test-integration` sets up these variables already [here](https://github.com/openshift/origin/blob/master/hack/test-integration.sh#L9). All temporary files should prefer to use `$BASETMPDIR` or `$ARTIFACT_DIR` for temporary files and only default to a system temp-dir if these variables are not set. You are solving the same problem.

Comment (php-coder) [inline test/integration/newapp_test.go]: I'm trying to reduce an amount of garbage that is left on my file system after running the tests. And what you're suggesting, it doesn't give it to me -- these scripts won't remove these temporary files. Also I want to have ability to run unit tests from my IDE and don't rely on our scripts at all.

Comment (php-coder) [inline test/integration/newapp_test.go]: P.S. Another try to explain my position: I'm fixing the fact that tests create files and don't remove them. For this task it's irrelevant where these files are created, it's a different story and I don't want to deal with that. If someone wants to fix it, let him do it in a separate PR.

Comment (stevekuznetsov) [inline test/integration/newapp_test.go]: Our scripts remove temporary files they know about -- those in `$BASETMPDIR`. Placing temporary files _outside_ of the normal place is a mistake and a bug. If they are placed within our configured temp directory, they will be cleaned up.

> Also I want to have ability to run unit tests from my IDE and don't rely on our scripts at all.

You can _also_ do this if you'd like.

Comment (stevekuznetsov) [inline test/integration/newapp_test.go]: One slight difference is that we have configured all of our tests to clean up at startup, not on teardown. This is to keep as much around as possible after a failed run. The only point at which we can be certain that some artifact will not be useful is once someone runs a new suite of tests. Otherwise, we could be deleting useful artifacts that may help someone debugging a flake or failure.

Comment (php-coder) [inline test/integration/newapp_test.go]: The files that we're talking about, are created in the tests and don't contains any useful info and they can be easily re-created again. It means that they should be always removed, because they're test data. And also they should be removed on the level where they were created.

@stevekuznetsov this change don't make our tests perfect but at least it makes them better than they're now. Are you agree?

Comment (php-coder) [inline test/integration/newapp_test.go]: What you talking looks like your personal opinion to me. And because of that I don't see why I should change my mind because your opinion equals to mine for me. It would be useful if you give me links with explanation of how we chose it should be. Also I see that integration tests create files not how you expect that they should do.

Comment (stevekuznetsov) [inline test/integration/newapp_test.go]: > What you're talking looks like your personal opinion to me.

What I am describing is how every single one of our test suites runs ... run a bunch of them and look at the tree under `/tmp/openshift`.

> It would be useful if you give me links with explanation of how we chose it should be.

I don't think we have a design document for something as simple as temporary directory file structure ... but you can look at the temporary directory setup [function](https://github.com/openshift/origin/blob/master/hack/lib/util/environment.sh#L115-L161) [called](https://github.com/openshift/origin/blob/master/hack/test-cmd.sh#L108) [in](https://github.com/openshift/origin/blob/master/hack/test-end-to-end-docker.sh#L13) [all](https://github.com/openshift/origin/blob/master/hack/test-end-to-end.sh#L46) [of](https://github.com/openshift/origin/blob/master/hack/test-go.sh#L42) [our](https://github.com/openshift/origin/blob/master/hack/test-integration.sh#L9) [test](https://github.com/openshift/origin/blob/master/test/extended/setup.sh#L36) [scripts](https://github.com/openshift/origin/blob/master/test/extended/ldap_groups.sh#L26) if you don't believe me... 

> Also I see that integration tests create files not how you expect that they should do.

And, again, I tell you this is incorrect behavior and a bug. I'm not asking you to fix all of these, but at least while you're changing this bit of code, make it place them in the right place.

Comment (php-coder) [inline test/integration/newapp_test.go]: Thanks, I'll take a look.

Do you agree that temp files can be removed from these tests? Because just changing the location doesn't delete them but only makes it easy to remove them manually later.

Comment (stevekuznetsov) [inline test/integration/newapp_test.go]: >  Because just changing the location doesn't delete them

When we place them under `$BASETMPDIR`, we can take advantage of the [code](https://github.com/openshift/origin/blob/master/hack/lib/util/environment.sh#L149-L159) that runs around our suites and cleans up there. If you believe these files would _never_ be useful to debug this test if it flakes out, then I guess we could delete them in Go as well, however, again, we have chosen to always cleanup the previous test's stuff on the next test's run just to make sure we don't accidentally get rid of useful artifacts. I'll defer to your judgment there but I will continue to prefer the cleanup-on-startup paradigm.

**Comment by php-coder** [comment]
@stevekuznetsov PTAL again: tests use `util.GetBaseDir()` now (and it uses `$BASETEMPDIR` under the hood). Also I'm removing test data at the end of tests only when they're succeeded.

**Comment by stevekuznetsov** [comment]
LGTM [test] this @openshift-bot

**Comment by stevekuznetsov** [comment]
[merge]
